webpack/JobWizard/steps/HostsAndInputs/buildHostQuery.js

```diff
@@ -1,12 +1,31 @@
 export const buildHostQuery = (selected, search) => {
+  const nameEscape = name => `"${name.replaceAll('"', '\\"')}"`;
   const { hosts, hostCollections, hostGroups } = selected;
-  const hostsSearch = `id ^ (${hosts.map(({ id }) => id).join(',')})`;
-  const hostCollectionsSearch = `host_collection_id ^ (${hostCollections
-    .map(({ id }) => id)
-    .join(',')})`;
-  const hostGroupsSearch = `hostgroup_id ^ (${hostGroups
-    .map(({ id }) => id)
-    .join(',')})`;
+  const MAX_NAME_ITEMS = 50;
+  let hostsSearch;
+  if (hosts.length < MAX_NAME_ITEMS)
+    hostsSearch = `name ^ (${hosts.map(({ name }) => name).join(',')})`;
+  else hostsSearch = `id ^ (${hosts.map(({ id }) => id).join(',')})`;
```

Comment:
Could we make this whole thing a bit dryer? Something along the lines of this, but applied everywhere?

```suggestion
  const buildSearch = (collection, search_key) => {
    const [real_search_key, attr] = (collection.length < MAX_NAME_ITEMS) ? [search_key, "name"] : ["id", "id"];
    let values = collection.map((item) => item[attr]);
    if (attr === "name") values = values.map((v) => nameEscape(v));
    return `${real_search_key} ^ (${values.join(',')})`;
  }
  
  hostsSearch = buildSearch(hosts, "name");
```

Reply:
its not really readable, can we not? especially since they live so close together and are only 3?

Reply:
I admit I went a bit overboard with code golfing there. 3 repetitions is where I usually draw the line, but I'm willing to let it slide this time